app/scripts/controllers/transactions/tx-state-manager.js

```diff
@@ -266,6 +266,11 @@ export default class TransactionStateManager extends EventEmitter {
     return txMeta;
   }
 
+  addExternalTransaction(txMeta) {
```

Comment:
One more request: could we add validation here to make sure that any external transaction added to state has a confirmed status and that it's nonce does not match the nonce of any current confirmed or pending transaction

Reply:
Sure.

Reply:
FYI, I added it yesterday, let me know what you think!